frontend/app/features/kanban/components/KanbanBoard.tsx

```diff
@@ -1,21 +1,87 @@
 "use client";
 
+import { useGetAllTasksQuery } from "@store/api/taskSlice";
+import { useState } from "react";
 import styles from "../styles/kanban.module.css";
+import { filterTasks, TaskFilters } from "../utils/filterTasks";
 import CreateTaskForm from "./CreateTaskForm";
+import TaskFilter from "./TaskFilter";
 
 const KanbanBoard: React.FC = () => {
+  const { data: tasks = [], error, isLoading } = useGetAllTasksQuery();
+  const [filters, setFilters] = useState<TaskFilters>({
+    status: "ALL",
+    createdBy: "ALL",
+    assignedTo: "ALL",
+    dueDate: "ALL",
+    createdAt: "ALL",
```

Comment:
please use react hook form for collecting filter data, and you do not have to use `ALL` for fetching all.

The logic should be, `IF` filter has some data, you pass it to BE and then BE filters it based on the values, but if there is no filter values, you fetch all data, so you only pass filters thats selected by user on FE to BE